logprep/ng/pipeline.py

```diff
@@ -0,0 +1,76 @@
+"""pipeline module for processing events through a series of processors."""
+
+from collections.abc import Iterator
+from itertools import islice
+from typing import Generator
+
+from logprep.ng.abc.event import Event
+from logprep.ng.abc.processor import Processor
+from logprep.ng.event.log_event import LogEvent
+
+
+class Pipeline(Iterator):
+    """Pipeline class to process events through a series of processors.
+    Examples:
+        >>> from logprep.ng.event.log_event import LogEvent
+        >>> from logprep.ng.abc.event import Event
+        >>> class MockProcessor:
+        ...     def process(self, event: LogEvent) -> None:
+        ...         event.data["processed"] = True
+        ...
+        >>>
+        >>> # Create test events
+        >>> events = [
+        ...     LogEvent({"message": "test1"}, original=b""),
+        ...     LogEvent({"message": "test2"}, original=b"")
+        ... ]
+        >>> processors = [MockProcessor()]
+        >>>
+        >>> # Create and run pipeline
+        >>> pipeline = Pipeline(iter(events), processors)
+        >>> processed_events = list(pipeline)
+        >>> len(processed_events)
+        2
+        >>> processed_events[0].data["processed"]
+        True
+        >>> processed_events[1].data["message"]
+        'test2'
+    """
+
+    def __init__(
+        self, input_connector: Iterator[Event], processors: list[Processor], process_count: int = 10
```

Comment:
Please use LogEvent as annotation to match with mehtod `_process_event` return type or vice versa.

```suggestion
        self, input_connector: Iterator[LogEvent], processors: list[Processor], process_count: int = 10
```

Reply:
This makes an unnecessary decision for the future. It is possible to put all `Events` here. We should not decide this for our future self ;)